.github/workflows/build.yml

```
@@ -0,0 +1,109 @@
+name: Build
+
+on:
+    push:
+        branches: [ master ]
+    pull_request:
+    release:
+        types: [ created ]
+
+jobs:
+    tests:
+        runs-on: ubuntu-latest
+        name: Build and test
+        strategy:
+            fail-fast: false
+            matrix:
+                include:
+                    -   php: 7.2
+                        tools: "composer:v1"
+                        git-fetch-depth: 1
+                    -   php: 7.3
+                        tools: "composer:v1"
+                        publish-phar: true
+                        git-fetch-depth: 0 # box/composer needs history to determine a recent git version
+                    -   php: 7.3
+                        tools: "composer:v1"
+                        phar-readonly: true
+                        git-fetch-depth: 1
+                    -   php: 7.4
+                        tools: "composer:v1"
+                        coverage: "pcov"
+                        git-fetch-depth: 1
+                    -   php: 7.4
+                        tools: "composer:v2"
+                        git-fetch-depth: 1
+
+        steps:
+            -   uses: actions/checkout@v2
+                with:
+                    fetch-depth: "${{ matrix.git-fetch-depth }}"
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: "${{ matrix.php }}"
+                    ini-values: "phar.readonly=0"
+                    tools: "${{ matrix.tools }}"
+                    coverage: "${{ matrix.coverage }}"
+                    extensions: ctype, iconv, xml
+
+            -   name: Get composer cache directory
+                id: composercache
+                run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            -   name: Cache composer dependencies
+                uses: actions/cache@v2
+                with:
+                    path: ${{ steps.composercache.outputs.dir }}
+                    key: ${{ runner.os }}-composer-cache-${{ hashFiles('**/composer.lock') }}
```

```suggestion
                    key: ${{ runner.os }}-composer-cache-${{ hashFiles('composer.*') }}
```
1. There isn't always a `composer.lock` (it is now, but wasn't for a long time)
2. No need to look for `composer.*` up the hierarchy, therefore no `**`

---

Done